acceptance/helper.rb

```diff
@@ -246,27 +246,68 @@ def sleep_until_started(host)
+    platform.include?('el-') or platform.include?('fedora')
+  end
+
+  def platform_is_deb_based(platform)
+    platform.include?('ubuntu') or platform.include?('debian')
+  end
+
+  def expected_version_for_platform(platform)
+    if platform_is_rpm_based(platform)
+      PuppetDBExtensions.config[:expected_rpm_version]
+    elsif platform_is_deb_based(platform)
+      PuppetDBExtensions.config[:expected_deb_version]
+    else
+      raise ArgumentError, "Unsupported platform: '#{platform}'"
+    end
+  end
+
 
+  def format_package_version_for_platform(platform, version)
     ## These 'platform' values come from the acceptance config files, so
     ## we're relying entirely on naming conventions here.  Would be nicer
     ## to do this using lsb_release or something, but...
-    if host['platform'].include?('el-5')
-      "#{PuppetDBExtensions.config[:expected_rpm_version]}.el5"
-    elsif host['platform'].include?('el-6')
-      "#{PuppetDBExtensions.config[:expected_rpm_version]}.el6"
-    elsif host['platform'].include?('el-7')
-      "#{PuppetDBExtensions.config[:expected_rpm_version]}.el7"
-    elsif host['platform'].include?('fedora')
-      version_tag = host['platform'].match(/^fedora-(\d+)/)[1]
-      "#{PuppetDBExtensions.config[:expected_rpm_version]}.fc#{version_tag}"
-    elsif host['platform'].include?('ubuntu') or host['platform'].include?('debian')
-      "#{PuppetDBExtensions.config[:expected_deb_version]}"
+    if platform.include?('el-5')
+      "#{version}.el5"
+    elsif platform.include?('el-6')
+      "#{version}.el6"
+    elsif platform.include?('el-7')
+      "#{version}.el7"
+    elsif platform.include?('fedora')
+      version_tag = platform.match(/^fedora-(\d+)/)[1]
+      "version.fc#{version_tag}"
+    elsif platform.include?('ubuntu') or platform.include?('debian')
+      # This already includes the '-1puppetlabs1' business
+      version
     else
-      raise ArgumentError, "Unsupported platform: '#{host['platform']}'"
+      raise ArgumentError, "Unsupported platform: '#{platform}'"
     end
+  end
 
+  def get_package_version(host, version = nil)
+    # Handle a few different kinds of input:
+    # - version = nil -> use PUPPETDB_EXPECTED_RPM_VERSION or PUPPETDB_EXPECTED_DEB_VERSION,
+    #                    depending on the distro. Add distro-specific formatting for rpm, but
+    #                    debian already inclues "-1puppetlabs1"
+    # - version = "x.y.z" -> Add distro-specific formatting to the given version, for both deb and rpm.
+    # - anything else: assume it's already been formatted correctly, just return it.
+
+    platform = host['platform']
+
+    if version.nil?
+      expected_version = expected_version_for_platform(platform)
+      return format_package_version_for_platform(platform, expected_version)
+    elsif version =~ /\d+\.\d+\.\d+/
+      actual_version = version
+      if platform_is_deb_based(platform)
+        actual_version += "-1puppetlabs1"
```

Comment:
Why not just have the deb clause add this if it doesn't already exist in the version string?